js/header.js

```diff
@@ -1,34 +1,50 @@
-import { renderWeekHeader} from './calendar.js'
-let currentDate = new Date();
+import { getFirstDayOfTheWeek, isToday } from './utils.js';
 
-function updateCurrentYearAndMonth(){
-    const currentMonthAndDaySpan = document.getElementById('currentMonthAndDay');
-    const options = { year: 'numeric', month: 'short' };
-    currentMonthAndDaySpan.textContent = currentDate.toLocaleDateString('en-US', options);
-}
-function switchWeeks(days) {
-    currentDate.setDate(currentDate.getDate() + days);
-    updateCurrentYearAndMonth();
-    renderWeekHeader(currentDate);
+export const renderWeekHeader = (date) => {
+  const weekHeader = document.querySelector(".week-header");
+  weekHeader.innerHTML = "";
+  const firstDayOfTheWeek = getFirstDayOfTheWeek(date);
+  const wholeWeekHeader = document.createElement("div");
+  wholeWeekHeader.className = "day";
+  for (let i = 0; i < 7; i++) {
+    const day = new Date(firstDayOfTheWeek);
+    day.setDate(day.getDate() + i);
+    const dayName = day.toLocaleDateString("en-US", { weekday: "short" });
+    const dayHeader = document.createElement("div");
+    dayHeader.className = "day";
+    dayHeader.innerHTML = `<span class='day-name'>${dayName}</span><span class="day-number">${day.getDate()}</span>`;
+    if (isToday(day)) {
+      dayHeader.classList.add("current-day");
+    }
+    weekHeader.appendChild(dayHeader);
+  }
+};
 
-}
+export const renderHeader = (date, updateDate) => {
+  const updateCurrentYearAndMonth = () => {
+    const currentMonthAndDaySpan = document.getElementById("currentMonthAndDay");
+    const options = { year: "numeric", month: "short" };
+    currentMonthAndDaySpan.textContent = date.toLocaleDateString("en-US", options);
+  };
 
-function eventListenerForSwitchingWeeksButton() {
-    document.getElementById('todayButton').addEventListener('click', () => {
-        currentDate = new Date();
-        switchWeeks(0);
-    });
+  const switchWeeks = (days) => {
+    const newDate = new Date(date);
```

Comment:
do you really need this? since date that comes as renderHeader param is already Date type

Reply:
In my opinion I do need this, I have to create a new object, because Date object is mutable (changing date object in switchWeeks results in general date change to the whole renderHeader). I tested, it doesn't work without creation of copy, doesn't render.